src/python/pants/invalidation/cache_manager.py

```diff
@@ -264,11 +287,13 @@ def __init__(self,
   def update(self, vts):
     """Mark a changed or invalidated VersionedTargetSet as successfully processed."""
     for vt in vts.versioned_targets:
+      vt._ensure_legal()
```

Comment:
This method should be public if it will be called here (it needn't be `API: public`)